stumpy/core.py

```diff
@@ -1675,6 +1683,9 @@ def _mass_distance_matrix(
                 T_subseq_isconstant,
             )
 
+            if query_idx is not None:
```

Comment:
I think at the top of this function, we should do:

```
if query_idx is not None:
    query_idx = int(query_idx)

for i in range(distance_matrix.shape[0]):
```

Reply:
Will add it to the code.